presto-main/src/main/java/com/facebook/presto/server/QueryResource.java

```diff
@@ -76,12 +77,14 @@ public QueryResource(DispatchManager dispatchManager, QueryManager queryManager)
 
     @GET
     @Path("{queryId}")
-    public Response getQueryInfo(@PathParam("queryId") QueryId queryId)
+    public Response getQueryInfo(@PathParam("queryId") QueryId queryId, @QueryParam("basic") @DefaultValue("false") boolean useBasic)
     {
         requireNonNull(queryId, "queryId is null");
-
         try {
             QueryInfo queryInfo = queryManager.getFullQueryInfo(queryId);
+            if (useBasic) {
+                return Response.ok(new BasicQueryInfo(queryInfo)).build();
```

Comment:
why not make call to  queryManager.getQueryInfo instread and skip the getFullQueryInfo call if the flag is true.